server/services/transaction.service.js

```diff
@@ -72,55 +74,50 @@ export const getPaymentDuration = async (offerId) => {
 };
 
 export const addTransaction = async (transaction) => {
-  const offer = await getOfferById(transaction.offerId).catch((error) => {
-    throw new ErrorHandler(httpStatus.INTERNAL_SERVER_ERROR, 'Internal Server Error', error);
+  const model = await getModelById({
+    modelType: transaction.model.type,
+    modelId: transaction.model.id,
   });
 
-  if (!offer) {
-    throw new ErrorHandler(httpStatus.NOT_FOUND, 'Invalid offer');
-  }
-
   try {
     const newTransaction = await new Transaction({
       ...transaction,
-      userId: offer.userId,
-      propertyId: offer.propertyId,
-      vendorId: offer.vendorId,
-      addedBy: offer.vendorId,
-      updatedBy: offer.vendorId,
+      vendorId: model.vendorId,
```

Comment:
This place is a little bit tricky, however, I think we need to confirm that `userId` exists first and use it here instead of vendorId (vendorId is not used by VAS).

```suggestion
      userId: model.userId,
```

For now, we might leave the propertyId and vendorId but will not make them required for now. When we get it right, maybe we will add another object and add all the related models, I am sure of the best solution for now. We need to think of a better way to make this work flawlessly. 

For `addedBy` and `updatedBy`, let's make them optional too. However, we need to use the Id of the user calling this service (if it is available).